### ISSUE TITLE
Warning when using @newtype in package object

### ISSUE THREAD

**Issue by gregghz** [estatico/scala-newtype#44]
I have the following code:

```
package object stuff {
  @newtype case class Thing(underlying: Int)
}
```

In scala 2.12.x the following warning is issued (making -Xfatal-warning impossible):

```
[warn] package.scala:10: it is not recommended to define classes/objects inside of package objects.
[warn] If possible, define trait MetricActorRef__Types in package metrics instead.
[warn]   @newtype case class MetricActorRef(underlying: ActorRef)
[warn]    ^
```

This warning happens if I use any scala-newtype version > `0.2.1` (that's the last version that works without the warning).

I am able to work around the issue by using https://github.com/ghik/silencer with the pathFilters pointing to the file in question (the @silent annotation does not work).

**Comment by gregghz** [comment]
This happens because the macro expands roughly like this:

```
package object stuff {
  trait Thing_Types { ... }
  object Thing extends AnyRef with Thing_Types { ... }
}
```

The trait that gets generated causes the warning. A simple solution is to wrap the trait in some project object or get rid of the trait altogether and just inline those members (but I'm not exactly sure the purpose of the trait).

**Comment by gregghz** [comment]
One other import point I left out above (because I didn't realize i had it set) is that this only happens when you have `-Xlint` enabled.

**Comment by gregghz** [comment]
>A simple solution is to wrap the trait in some proxy object

It turns out this doesn't work. It just changes the warning to "top-level class without companion can only expand either into an eponymous class or into a block consisting in eponymous companions"

**Comment by joroKr21** [comment]
The trait is there to make the encoding work on Scala 2.10 (it doesn't allow abstract types in objects). Maybe with the release of Scala 2.13 support for 2.10 can be dropped and the trait removed.

**Comment by gregghz** [comment]
ah, interesting. Do you think it would be acceptable using https://github.com/sbt/sbt-buildinfo to detect the compiler version and only add the trait when scala 2.10 is detected? If so, I could probably work that out.

**Comment by joroKr21** [comment]
Actually this library already uses version-dependent sources. Look for `NewTypeCompatMacros.scala`. You could just add a flag there `emitTrait: Boolean` or whatever and conditionally generate the trait.

I think it's definitely worth doing that, not just because of the warning but also to reduce the overhead in terms of generated class files. :+1:

**Comment by carymrobbins** [comment]
Fixed via #50.
